app/services/usps_confirmation_uploader.rb

```diff
@@ -7,8 +7,8 @@ def run
     confirmations = UspsConfirmation.all
```

Comment:
I think we should consider a `.to_a` on here:

```suggestion
    confirmations = UspsConfirmation.all.to_a
```

which gives us more deterministic behavior